src/EventSubscriber/InitSubscriber.php

```diff
@@ -138,8 +141,8 @@ public function checkRequiredPasswordChange(GetResponseEvent $event): void {
    */
   public function passwordChangeNotification(GetResponseEvent $event): void {
     if ($event->isMasterRequest() && Url::fromRoute('password_enhancements.password_change')->toString() !== $event->getRequest()->getPathInfo()) {
-      $policy = $this->policyStorage->loadByRoleAndPriority($this->account->getRoles());
-      if ($policy !== NULL && $this->passwordChecker->showWarningMessage($policy)) {
+      $policy = $this->policyManager->loadPolicyByRoles($this->account->getRoles());
+      if ($policy !== NULL && $this->passwordChecker->isWarningMessageNeeded($policy)) {
```

Comment:
` '@url' => $user->toUrl('edit-form')->toString(),`
instead of
` '@url' => Url::fromRoute('entity.user.edit_form', ['user' => $user->id()])->toString(),`